src/l500/l500-options.cpp

```diff
@@ -227,22 +336,27 @@ namespace librealsense
                                                   _hw_monitor.get(),
                                                   invalidation_bypass,
                                                   resolution_option.get(),
-                                                  "Enable/disable pixel invalidation" );
+                                                  "Enable/disable pixel invalidation",
+                                                  _fw_version,
+                                                  _digital_gain );
 
-            _digital_gain = register_option<uvc_xu_option<int>, uvc_sensor&, platform::extension_unit, uint8_t, std::string, const std::map<float, std::string>& >
-                (RS2_OPTION_DIGITAL_GAIN, raw_depth_sensor, ivcam2::depth_xu, ivcam2::L500_DIGITAL_GAIN,
-                    "Change the depth digital gain to: 1 for high gain and 2 for low gain",
-                    std::map<float, std::string>{ { RS2_DIGITAL_GAIN_HIGH, "High Gain"},
-                    { RS2_DIGITAL_GAIN_LOW, "Low Gain" }});
+             auto preset = calc_preset_from_controls();
 
+             _preset = std::make_shared< l500_preset_option >(
+                 option_range{ RS2_L500_VISUAL_PRESET_CUSTOM,
+                               RS2_L500_VISUAL_PRESET_SHORT_RANGE,
+                               1,
+                               (float)preset },
+                 "Preset to calibrate the camera to environment ambient, no ambient or low "
+                 "ambient.",
+                 this );
 
-            _preset = register_option <float_option_with_description<rs2_l500_visual_preset>, option_range>
-                (RS2_OPTION_VISUAL_PRESET, option_range{ RS2_L500_VISUAL_PRESET_CUSTOM , RS2_L500_VISUAL_PRESET_SHORT_RANGE, 1, RS2_L500_VISUAL_PRESET_DEFAULT },
-                    "Preset to calibrate the camera to environment ambient, no ambient or low ambient.");
+            depth_sensor.register_option( RS2_OPTION_VISUAL_PRESET, _preset );
 
-            _advanced_options = get_advanced_controls();
-        }
+        set_preset_controls( preset );
```

Comment:
We just calculated the preset from the controls... why do we need to now set the controls from the preset?